rust-runtime/aws-smithy-async/src/rt/sleep.rs

```diff
@@ -81,26 +90,66 @@ impl Future for Sleep {
 
 /// Implementation of [`AsyncSleep`] for Tokio.
 #[non_exhaustive]
-#[cfg(feature = "rt-tokio")]
+#[cfg(all(
+    feature = "rt-tokio",
+    not(all(target_family = "wasm", target_os = "wasi"))
+))]
 #[derive(Debug, Default)]
 pub struct TokioSleep;
```

Comment:
we really need to avoid complex feature gating like this

Reply:
removed